src/core/TextSelector.ts

```diff
@@ -131,6 +132,38 @@ export default class TextSelector {
     return ranges;
   }
 
+  currentSerializedSelection = () => {
+    const selectedRanges = this.captureDocumentSelection();
+    return this.serializeRanges(selectedRanges);
+  }
+
+  serializeRanges = (ranges: Range.NormalizedRange[]) => {
```

Comment:
if you'd like this function to be super slick (😛), you could write it like:
```
return ranges.reduce((acc, range) => acc.push(range.serialize(this.element, `.${PPHighlightClass}`)), []);
```

or even better:
```
ranges.map(range => range.serialize(this.element, `.${PPHighlighClass}`);
```

Reply:
Actually, apart from one-liner tricks, these functions can have a bit better performance. 

Assuming a large collection in `ranges`, the `for` loop actually blocks the js thread for its whole duration (possibly freezing the UI).

`reduce` and `map` on the other hand are callback-based, so each call(or chunks of them) are done in separate events in the event loop, allowing the UI to work normally in the meantime

Reply:
On the other hand - I guess `selectedRanges` (line 193) is never going to be large? IIRC it's usually 1 element?

Reply:
Oh, I didn't know about this blocking/non-blocking difference. It's good to know.

But yes, unforutantely it's true that we more or less asume there won't be more than 1 annotation range :)